packages/storage/src/reference.ts

```diff
@@ -289,8 +338,8 @@ export async function getMetadata(ref: Reference): Promise<Metadata> {
  *     See `firebaseStorage.Reference.prototype.getMetadata`
  */
 export async function updateMetadata(
-  ref: Reference,
-  metadata: Metadata
+  ref: StorageReference,
+  metadata: Record<string, unknown>
```

Comment:
Isn't Metadata more than just a `Record<string, unknown>` since some keys are pre-defined? It seems like changing this to a Record loses some information, even if it doesn't make a huge difference as an input type.

Reply:
Metadata requires all its fields, the param that can be supplied here only needs to have some of the fields, or even just 1. Should it be named something else? `metadataUpdates`?

Reply:
What about `Partial<Metadata>`? I think that makes the intent very clear.

Reply:
Oh, perfect, done.